FileSystem/FileIO.js

```diff
@@ -0,0 +1,254 @@
+  lof: lof,
+  isEOF: isEOF,
+  printToFile: printToFile,
+
+  openFiles: {},
+
+  closeFile: closeFile,
+};
+
+/**
+ * Emulates VBA open statement API
+ * @todo Implement file locking
+ * @body File locking is not possible in Google Drive but it could be emulated by using PropertyServices.
+ * @param {string} path The local file path
+ * @param {number} fileNumber File number
+ * @param {string} openMode Open Mode enumeration
+ * @param {string} accessMode Access Mode enumeration
+ * @param {string} lockMode lockMode enumeration
+ */
+function openFile(path, fileNumber, openMode, accessMode, lockMode) {
+  // Set default arguments
+  openMode = openMode || OpenMode.RANDOM;
+  accessMode = accessMode || AccessMode.READ_WRITE;
+  lockMode = lockMode || LockMode.NONE;
+
+  // Check if file number is available
+  if (fileNumber in this.openFiles) {
+    throw Error('File Number: ' + fileNumber + ' is already being used');
+  }
+
+  // Check if file number is valid
+  if (fileNumber < 1 || fileNumber > 511) {
+    throw Error(
+      'File Number: ' +
+        fileNumber +
+        ' is invalid. File numbers need to be an number value between 1 and 511'
+    );
+  }
+
+  var fileId;
+
+  // If file exists, get file id else create and get file id
+  if (FileMapper.hasMapping(FileSystem.currentDirectory, path)) {
+    fileId = FileMapper.getFileId(FileSystem.currentDirectory, path);
+  } else {
+    fileId = createFile(FileSystem.currentDirectory, path, MimeType.PLAIN_TEXT);
+  }
+
+  // In memory file object
+  var file = {
+    fileId: fileId,
+    filePath: path,
+    accessMode: accessMode,
+    openMode: openMode,
+    lockMode: lockMode,
+  };
+
+  // Set file content (In memory buffer)
+  switch (openMode) {
+    case OpenMode.INPUT:
+    case OpenMode.APPEND:
+    case OpenMode.RANDOM:
+      file.content = DriveApp.getFileById(file.fileId)
+        .getBlob()
+        .getDataAsString();
+      break;
+    case OpenMode.OUTPUT:
+      file.content = '';
+      break;
+    case OpenMode.BINARY:
+      file.content = DriveApp.getFileById(file.fileId).getBlob().getBytes();
+      break;
+  }
+
+  // Set file pointer
+  switch (openMode) {
+    case OpenMode.INPUT:
+    case OpenMode.RANDOM:
+    case OpenMode.OUTPUT:
+    case OpenMode.BINARY:
+      file.pointer = 0;
+      break;
+    case OpenMode.APPEND:
+      file.pointer = file.content.length;
+      break;
+  }
+
+  // Register Open File
+  this.openFiles[fileNumber] = file;
+}
+
+/**
+ * Internal method to close file
+ * Saves the file content and closes the removed the file
+ * @param {number} fileNumber File number to be closed
+ */
+function closeFile(fileNumber) {
+  if (!(fileNumber in this.openFiles)) {
+    throw Error('File Number: ' + fileNumber + ' is not open');
+  }
+
+  var file = this.openFiles[fileNumber];
+
+  if (file.openMode == OpenMode.BINARY) {
+    DriveApp.getFileById(file.fileId).getBlob().setBytes(file.content); // Does this work??
```

Comment:
Does this work?

Reply:
Haven't tested because haven't implemented binary file IO. Should I remove this?

Reply:
How about TODO?

Reply:
Done